### PR TITLE
feat: Add WORKFLOW_MAX_EXECUTION_TIME env var

### PR DESCRIPTION
**Pull request by majian159**
…vironment variable `WORKFLOW_MAX_EXECUTION_TIME`

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/core/workflow/workflow_engine_manager.py [hunk 1]
```diff
@@ -30,6 +31,8 @@
     WorkflowNodeExecutionStatus,
 )
 
+WORKFLOW_MAX_EXECUTION_TIME = int(os.environ.get('WORKFLOW_MAX_EXECUTION_TIME', str(600)))  # 600s
```
Comment (takatost) [inline api/core/workflow/workflow_engine_manager.py]: Suggest putting this variable in config.py and accessing it through current_app.config.get().

File: api/core/workflow/workflow_engine_manager.py [hunk 2]
```diff
@@ -153,7 +156,7 @@ def run_workflow(self, workflow: Workflow,
                     raise ValueError('Max steps 50 reached.')
 
                 # or max execution time 10min reached
-                if self._is_timed_out(start_at=workflow_run_state.start_at, max_execution_time=600):
+                if self._is_timed_out(start_at=workflow_run_state.start_at, max_execution_time=WORKFLOW_MAX_EXECUTION_TIME):
                     raise ValueError('Max execution time 10min reached.')
```
Comment (takatost) [inline api/core/workflow/workflow_engine_manager.py]: And recommended to keep the max_execution_time value in the error message consistent with the value of the environment variable.

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/workflow/workflow_engine_manager.py (hunk 1 above)
Comment (majian159) [reply api/core/workflow/workflow_engine_manager.py]: fix

File: api/core/workflow/workflow_engine_manager.py (hunk 2 above)
Comment (majian159) [reply api/core/workflow/workflow_engine_manager.py]: fix

**Review by takatost — DISMISSED** [review]
LGTM

**Review by takatost — APPROVED** [review]

**Pull request merged**